controllers/workspace/devworkspace_controller.go

```diff
@@ -279,7 +293,13 @@ func (r *DevWorkspaceReconciler) Reconcile(req ctrl.Request) (reconcileResult ct
 	deploymentStatus := provision.SyncDeploymentToCluster(workspace, allPodAdditions, serviceAcctName, clusterAPI)
 	if !deploymentStatus.Continue {
 		if deploymentStatus.FailStartup {
-			return r.failWorkspace(workspace, fmt.Sprintf("DevWorkspace spec is invalid: %s", deploymentStatus.Err), reqLogger, &reconcileStatus)
+			var message string
+			if deploymentStatus.Err != nil {
```

Comment:
Why error has a higher priority? Isn't the message supposed to be user-friendly?

Reply:
Yeah, sometimes SynDeploymentToCluster doesn't return an error though, so doing deploymentStatus.Err.Error() will fail because Err is nil

Reply:
With the current codebase it should work since we return error or message.

But I think in case both are returned, we need to do one of the following
1. Mix message and error in the condition message, as we do with ProvisioningError. Actually, what if we change ProvisioningStatus to have `ProvisioningError` instead of Err + Message? Then it will work the same.
2. Or message has higher priority to be provided into status message, error is used only when "user-friendly" is not specified.